pandas/core/indexes/datetimes.py

```diff
@@ -227,11 +227,11 @@ def __new__(cls, data=None,
                           "endpoints is deprecated.  Use "
                           "`pandas.date_range` instead.",
                           FutureWarning, stacklevel=2)
-            result = cls._generate_range(start, end, periods,
-                                         freq=freq, tz=tz, normalize=normalize,
-                                         closed=closed, ambiguous=ambiguous)
-            result.name = name
-            return result
+            dtarr = DatetimeArray._generate_range(
+                start, end, periods,
+                freq=freq, tz=tz, normalize=normalize,
+                closed=closed, ambiguous=ambiguous)
+            return cls(dtarr, name=name)
```

Comment:
I think in #24024 this becomes `cls._simple_new(dtarr, name=name)`.  That isn't possible yet here.

Reply:
ditto below and for TimedeltaIndex

Reply:
Is it worth doing this now then, when we'll just need to change it again?